service/worker/replicator/replicator.go

```diff
@@ -172,3 +182,64 @@ func (r *Replicator) listenToClusterMetadataChange() {
 		},
 	)
 }
+
+func (r *Replicator) cleanupAckedMessages(
+	ctx context.Context,
+	ackMessageID int64,
+) (int64, error) {
+	ackLevelByCluster, err := r.namespaceReplicationQueue.GetAckLevels(ctx)
+	if err != nil {
+		return ackMessageID, err
+	}
+
+	connectedClusters := r.clusterMetadata.GetAllClusterInfo()
+	maxAckLevel := ackMessageID
+	minAckLevel := int64(math.MaxInt64)
+	for clusterName, ackLevel := range ackLevelByCluster {
+		if clusterName == r.clusterMetadata.GetCurrentClusterName() {
+			continue
+		}
+		if ackLevel > maxAckLevel {
+			maxAckLevel = ackLevel
+		}
+		if _, ok := connectedClusters[clusterName]; !ok {
+			continue
```

Comment:
when do we delete ack level for disconnected clusters?

Reply:
No in this PR. Do you want to delete the ack levels? Or we just migrate to queue v2?

Reply:
Added a TODO, I need to expose new persistence API to support this.